oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoVersionGCSupport.java

```diff
@@ -132,6 +133,49 @@ public CloseableIterable<NodeDocument> getPossiblyDeletedDocs(final long fromMod
                 input -> store.convertFromDBObject(NODES, input)));
     }
 
+    /**
+     * Calculate the bson representing including only the provided
+     * include path prefixes and/or excluding the provided
+     * exclude path prefixes - if any are provided - AND the provided
+     * query
+     * @param includes set of path prefixes which should only be considered
+     * @param excludes set of path prefixes which should be excluded.
+     * if these overlap with includes, then exclude has precedence.
+     * @param query the query with which to do an AND
+     * @return the combined bson with include/exclude path prefixes
+     * AND the provided query
+     */
+    private Bson withIncludeExcludes(Set<String> includes, Set<String> excludes, Bson query) {
+        Bson inclExcl = null;
+        if (includes != null && !includes.isEmpty()) {
+            final List<Bson> ors = new ArrayList<>(includes.size());
+            for (String incl : includes) {
+                ors.add(Filters.regex(ID, ":" + incl));
```

Comment:
Don't we need to add `level` before `:` ?

Reply:
the idea is that any string before `:` matches. So any level would be fine. But also any split document in theory - except they don't get included in the outer query as they have no `_modified`.